source/common/router/config_impl.h

```diff
@@ -86,6 +88,8 @@ class VirtualHostImpl : public VirtualHost {
   const std::string& name() const override { return name_; }
   const RateLimitPolicy& rateLimitPolicy() const override { return rate_limit_policy_; }
 
+  RequestHeaderParser& requestHeaderParser() const { return *request_headers_parser_; };
```

Comment:
I find it strange that the method is declared as `const` but returns a reference to internal data.
I'd remove `const` in here.